app.py

```diff
@@ -72,13 +79,37 @@ def analyze_document():
     # dict.pop('tables')
     # dict.pop('paragraphs')
     # form = language_model.generate_form(json.dumps(dict))
-    form = language_model.generate_form(dict['content'])
     
-    return app.response_class(
-        response=form,
-        status=200,
-        mimetype='application/json'
-    )
+    global data
+    data = language_model.generate_form(dict['content'])
+
+@app.route('/analyze', methods=['GET'])
+def analyze_document():
+    
+    document = document_storage.get_document()
+    if not document:
+        return "No documents to analyze", 400
+    
+    
+    
+    print(f"data from route pov : {data}")
+    if data == "":
+        if not SHOULD_GET_DATA :
```

Comment:
Wouldn't this lock any concurrent request to the backend?

Reply:
Yes it does,
I believe there are other issues with concurrent request to the backend
Might all be solved with user accounts later